### PR TITLE
ref(streams): Improve processing submodule layout

### PR DESCRIPTION
**Pull request by tkaemming**
Similar energy to #1304. Here is a summary of the new layout:

```
snuba.utils.streams.processing.
    strategies.
        abstract.
            ProcessingStrategy
            ProcessingStrategyFactory
        batching.
            AbstractBatchWorker
            BatchProcessingStrategy
            BatchProcessingStrategyFactory
        streaming.
            collect.
                CollectStep
            filter.
                FilterStep
            transform.
                TransformStep
                ParallelTransformStep
    processor.
        StreamProcessor
```

This should make the streams module easier to understand/navigate, and also will improve the readability of profiler stats and stack traces, as components like the streaming steps will be referenced by a more component specific file path instead of generically referencing "poll from the streaming package".

### CODE REVIEW

File: snuba/cli/replacer.py [hunk 1]
```diff
@@ -84,8 +84,10 @@ def replacer(
         TransportError,
         build_kafka_consumer_configuration,
     )
-    from snuba.utils.streams.batching import BatchProcessingStrategyFactory
-    from snuba.utils.streams.processing import StreamProcessor
+    from snuba.utils.streams.processing.processor import StreamProcessor
```
Comment (fpacifici) [inline snuba/cli/replacer.py:87]: Any reason not to expose StreamProcessor directly in `snuba.utils.stresms.processing` ? It seems to me that additional `processor` module is unlikely to contain much else that should be exported.

File: snuba/utils/streams/processing/processor.py [hunk 2]
```diff
@@ -2,123 +2,19 @@
...
-    Type,
-)
+from typing import Generic, Mapping, Optional, Sequence, Type
 
 from snuba.utils.streams.backends.abstract import Consumer, ConsumerError
+from snuba.utils.streams.processing.strategies.abstract import (
+    MessageRejected,
+    ProcessingStrategy,
+    ProcessingStrategyFactory,
+)
```
Comment (fpacifici) [inline snuba/utils/streams/processing/processor.py:12]: export this in `snuba.utils.streams.processing.strategies` ?

**Review by fpacifici — APPROVED** [review]
I would consider taking the stream package out of utils entirely. I think having that `utils` does not provide much value, especially since now streams are a major subsystem. But this is probably out of scope.

**Comment by tkaemming** [comment]
Oops, oversight on the import aliasing — will clean that up.

> I would consider taking the stream package out of utils entirely. I think having that `utils` does not provide much value, especially since now streams are a major subsystem. But this is probably out of scope.

The only hesitation that I have with that kind of a change is that to me, the contents within `streams` doesn't seem to have the same importance or operate at the same level of abstraction as many of the other packages in the `snuba.*` namespace. It only exists to support the implementation of those other packages and doesn't have much value on it's own outside of that context. I do agree that `utils` kind of hides the magnitude of the module (it's kind of like calling an excavator a "power tool"… it's not wrong, but it's not really representative either.) There's probably a better compromise here, but I can't think of one off of the top of my head.

**Comment by tkaemming** [comment]
> There's probably a better compromise here, but I can't think of one off of the top of my head.

Maybe something like `snuba.services` that includes cache, metrics, streams, etc.?

**Pull request merged**